src/gpu/generic/sycl/pooling_kernels.hpp

```diff
@@ -377,104 +399,39 @@ struct pooling_bwd_kernel_vec_t {
         return 0;
     }
 
-    void ker_zero(out_memory_tensor_t &diff_src_mem, dim_t mb, dim_t oc) const {
-        for_(dim_t id = 0; id < conf_.ID; ++id)
-        for_(dim_t ih = 0; ih < conf_.IH; ++ih)
-        for (dim_t iw = 0; iw < conf_.IW; ++iw) {
-            const auto off = get_offset(diff_src_md(), mb, oc, id, ih, iw);
-            diff_src_mem.store(0, off);
-        }
-    }
-
-    void ker_max(out_memory_tensor_t &diff_src_mem,
-            const in_memory_tensor_t &diff_dst_mem, dim_t mb, dim_t oc,
-            dim_t od, dim_t oh, dim_t ow) const {
-        memory_tensor_t ws_mem(ws_, conf_.ws_md);
-        const auto ws_off = get_offset(ws_md(), mb, oc, od, oh, ow);
-
-        const int index = ws_mem.load(ws_off);
-        const dim_t kd = (index / conf_.KW) / conf_.KH;
-        const dim_t kh = (index / conf_.KW) % conf_.KH;
-        const dim_t kw = index % conf_.KW;
-        const dim_t id = od * conf_.SD - conf_.padF + kd * (conf_.DD + 1);
-        const dim_t ih = oh * conf_.SH - conf_.padT + kh * (conf_.DH + 1);
-        const dim_t iw = ow * conf_.SW - conf_.padL + kw * (conf_.DW + 1);
-        if (id < 0 || id >= conf_.ID) return;
-        if (ih < 0 || ih >= conf_.IH) return;
-        if (iw < 0 || iw >= conf_.IW) return;
-
-        const auto d_src_off = get_offset(diff_src_md(), mb, oc, id, ih, iw);
-        const auto d_dst_off = get_offset(diff_dst_md(), mb, oc, od, oh, ow);
-        float v_src = diff_src_mem.load(d_src_off);
-        float v_dst = diff_dst_mem.load(d_dst_off);
-        v_src += v_dst;
-        diff_src_mem.store(v_src, d_src_off);
```

Comment:
@dzarukin as you mentioned [here](https://github.com/oneapi-src/oneDNN/pull/2039#discussion_r1725232416) there was an issue about the accumulation stage. The values were being loaded and stored continuously, making the accumulation not work as expected. I fixed these issues in the current commit. I accumulated the elements and finally stored them at the end. I refactored the code to make it functional. I run the benchdnn tests on Nvidia and also Intel PVC.

Reply:
Thank you for addressing it.